articles/quickstart/spa/angular-beta/02-calling-an-api.md

```diff
@@ -0,0 +1,284 @@
+---
+title: Calling an API
+description: This tutorial demonstrates how to call a backend API using an access token.
+budicon: 448
+topics:
+  - quickstarts
+  - spa
+  - angular
+  - login
+github:
+    path: 02-Calling-an-API
+contentType: tutorial
+useCase: quickstart
+---
+
+<!-- markdownlint-disable MD006 MD002 MD041 -->
+
+Most single-page apps use resources from data APIs. You may want to restrict access to those resources, so that only authenticated users with sufficient privileges can access them. Auth0 lets you manage access to these resources using [API Authorization](/api-auth).
+
+This tutorial shows you how to create a simple API using [Express](https://expressjs.com) that validates incoming JSON Web Tokens. You will then see how to call this API using an Access Token granted by the Auth0 authorization server.
+
+<%= include('../_includes/_calling_api_create_api') %>
+
+<%= include('../_includes/_calling_api_create_backend.md') %>
+
+Finally, modify `package.json` to add two new scripts `dev` and `server` that can be used to start the frontend and the backend API together:
+
+```json
+"scripts": {
+  "ng": "ng",
+  "start": "ng serve",
+  "build": "ng build",
+  "test": "ng test",
+  "lint": "ng lint",
+  "e2e": "ng e2e",
+  "server": "node server.js",
+  "dev": "npm-run-all --parallel start server"
+},
+```
+
+To start the project for this sample, run the `dev` command from the terminal:
+
+```bash
+npm run dev
+```
+
+This will start both the backend API and the frontend application together.
+
+### Set up a proxy to the backend API
+
+In order to call the API from the frontend application, the development server must be configured to proxy requests through to the backend API. To do this, add a `proxy.conf.json` file to the root of the project and populate it with the following code:
```

Comment:
> ... the development server must be configured to proxy requests through to the backend API.

This explanation is a little confusing. Why are we doing that? Is this more for the example here or is this a common real-world task?

Reply:
I've reworded the text here, hopefully it's more clear.

But this is one of those things that is unnecessary for the authentication to work but is kind of necessary for the sample. We could configure the server with CORS so that the front-end running on `http://localhost:3000` can just call `http://localhost:3001`, then we don't need the proxy config and is actually probably a more real-world example anyway.